base/est/src/main/java/org/dogtagpki/est/ESTEngine.java

```diff
@@ -0,0 +1,105 @@
+package org.dogtagpki.est;
+
+import java.io.File;
+import java.io.FileReader;
+import java.util.Properties;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+/**
+ * Engine that manages the EST backend(s) according to configuration.
+ *
+ * @author Fraser Tweedale
+ */
+@WebListener
+public class ESTEngine implements ServletContextListener {
+
+    private static org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(ESTEngine.class);
+
+    private static ESTEngine INSTANCE;
+
+    private ESTBackend backend;
+
+    public static ESTEngine getInstance() {
+        return INSTANCE;
+    }
+
+    public ESTEngine() {
+        INSTANCE = this;
```

Comment:
`INSTANCE` can be reassigned. If this has to be a singleton than some additional checks are required or if available exploit the engine (I am sure EJB support singleton annotation but I do not know if available in tomcat).

Reply:
Good point.  I will find a more robust way.